drivers/i2c/i2c_ll_stm32.c

```diff
@@ -313,4 +313,47 @@ static void i2c_stm32_irq_config_func_3(struct device *dev)
 
 #endif /* CONFIG_I2C_3 */
 
+#ifdef CONFIG_I2C_4
+
+#ifndef I2C4_BASE
+#error "I2C_4 is not available on the platform that you selected"
+#endif /* I2C4_BASE */
+
+#ifdef CONFIG_I2C_STM32_INTERRUPT
+static void i2c_stm32_irq_config_func_4(struct device *port);
+#endif
+
+static const struct i2c_stm32_config i2c_stm32_cfg_4 = {
+	.i2c = (I2C_TypeDef *)CONFIG_I2C_4_BASE_ADDRESS,
+	.pclken = {
+		.enr = LL_APB1_GRP2_PERIPH_I2C4,
```

Comment:
On STM32F7 series it is LL_APB1_GRP1_PERIPH_I2C4. Could you add an #ifdef here?

Reply:
F7 is not yet ported. I prefer to keep it separate.
Even if it should arrive rather soon, this PR might be merged in 1.12 while F7 will likely be part of 1.13

Reply:
To make myself clear clear, I was talking about the following #ifdef which allows the PR to be merged in 1.12:
```
#ifdef SOC_SERIES_STM32L4X
         .enr = LL_APB1_GRP2_PERIPH_I2C4,
         .bus = STM32_CLOCK_BUS_APB1_2,
#else
         .enr = LL_APB1_GRP1_PERIPH_I2C4,
         .bus = STM32_CLOCK_BUS_APB1,
#endif
```